libraries/redcore/api/hal/hal.php

```diff
@@ -628,6 +628,26 @@ public function apiCreate()
 		$functionName = RApiHalHelper::attributeToString($this->operationConfiguration, 'functionName', 'save');
 		$data = $this->triggerFunction('processPostData', $this->options->get('data', array()), $this->operationConfiguration);
 
+		// Remove primary keys for "model" and "table" data mode
+		$dataMode = RApiHalHelper::attributeToString($this->operationConfiguration, 'dataMode');
+
+		if (($dataMode == 'model') || ($dataMode == 'table'))
+		{
+			// Get the primary keys
+			$primaryKeys = $model->getTable()->getKeyName(true);
```

Comment:
Actually on a second thought about this, if more than one PK field is defined, this can be problematic, because two fields cannot be auto-incremental.
I think in any case we should unset the auto-incremental fields only.